packages/push/src/instruction/common/dup.rs

```diff
@@ -0,0 +1,184 @@
+use std::marker::PhantomData;
+
+use super::super::{Instruction, instruction_error::PushInstructionError};
+use crate::{
+    error::InstructionResult,
+    push_vm::{HasStack, stack::PushOnto},
+};
+
+/// An instruction that duplicates (clones) the top
+/// value of a stack of type `T` in the given state.
+///
+/// # Inputs
+///
+/// The `Dup` instruction takes the following inputs:
+///    - Stack of type `T`
+///      - One value
+///
+/// # Behavior
+///
+/// The `Dup` instruction clones the top block on the stack of type `T`,
+/// leaving both the original and the copy on that stack.
+///
+/// If stack is empty, this is a no-op,
+/// and the state is returned unchanged.
```

Comment:
We actually return a recoverable `Underflow` error so idk how accurate this comment is.

Reply:
That's a good catch, and I think it's a actually a problem on most of the documentation. I'll fix it here, and I'll make an issue to fix it everywhere else.